libraries/chain/steem_evaluator.cpp

```diff
@@ -613,14 +613,24 @@ void comment_evaluator::do_apply( const comment_operation& o )
          if( o.parent_author == STEEM_ROOT_POST_PARENT )
              FC_ASSERT( ( now - auth.last_root_post ) > STEEM_MIN_ROOT_COMMENT_INTERVAL, "You may only post once every 5 minutes.", ("now",now)("last_root_post", auth.last_root_post) );
          else
-             FC_ASSERT( (now - auth.last_post) > STEEM_MIN_REPLY_INTERVAL, "You may only comment once every 20 seconds.", ("now",now)("auth.last_post",auth.last_post) );
+         {
+            if( _db.has_hardfork( STEEM_HARDFORK_0_20__2019 ) )
```

Comment:
Add a todo to remove this check after HF 20. Because we are reducing the interval from 20 to 3, after HF 20 we can do all old checks as 3 seconds and they will pass. Anything to help keep the codebase clean!